xlators/protocol/server/src/server-rpc-fops_v2.c

```diff
@@ -4360,6 +4419,17 @@ server4_0_setxattr(rpcsvc_request_t *req)
         goto out;
     }
 
+    /* Let the namespace setting can happen only from special mounts, this
+       should prevent all mounts creating fake namespace. */
+    if ((frame->root->pid >= 0) &&
+        dict_get_sizen(state->dict, GF_NAMESPACE_KEY)) {
+        gf_smsg("server", GF_LOG_ERROR, 0, PS_MSG_SETXATTR_INFO, "path=%s",
+                state->loc.path, "key=%s", GF_NAMESPACE_KEY, NULL);
+        ret = -1;
+        SERVER_REQ_SET_ERROR(req, ret);
+        goto out;
+    }
```

Comment:
What should happen in removexattr?

Reply:
as explained in another comment, haven't done removexattr() as it involves doing things beyond just one inode scope.

Reply:
In that case, until we handle it, do you want to give some errno like ENOTSUP or something?

Reply:
Done. Please check if thats fine. added a test case for that. Will run the tests anyways.